test/provisioner/provision.go

```diff
@@ -114,6 +217,165 @@ func GetInstallOverlay(provider string, installDir string) (InstallOverlay, erro
 	return overlayFunc(installDir)
 }
 
+func NewKbsInstallOverlay(installDir string) (InstallOverlay, error) {
+	log.Info("Creating kbs install overlay")
+	overlay, err := NewKustomizeOverlay(filepath.Join(installDir, "config/kubernetes/base"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &KbsInstallOverlay{
+		overlay: overlay,
+	}, nil
+}
+
+func (lio *KbsInstallOverlay) Apply(ctx context.Context, cfg *envconf.Config) error {
+	return lio.overlay.Apply(ctx, cfg)
+}
+
+func (lio *KbsInstallOverlay) Delete(ctx context.Context, cfg *envconf.Config) error {
+	return lio.overlay.Delete(ctx, cfg)
+}
+
+func (lio *KbsInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config, props map[string]string) error {
+	var err error
+	log.Infof("Updating kbs image with %q", props["KBS_IMAGE"])
+	if err = lio.overlay.SetKustomizeImage("kbs-container-image", "newName", props["KBS_IMAGE"]); err != nil {
+		return err
+	}
+
+	log.Infof("Updating CAA image tag with %q", props["KBS_IMAGE_TAG"])
+	if err = lio.overlay.SetKustomizeImage("kbs-container-image", "newTag", props["KBS_IMAGE_TAG"]); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *KeyBrokerService) GetKbsPodIP(ctx context.Context, cfg *envconf.Config) (string, error) {
```

Comment:
Do you want to deploy a  KBS service and use it's IP instead of getting the pod ip ?

Reply:
Yes that can be done as well, will make that change.